flink-kubernetes-operator-autoscaler/src/main/java/org/apache/flink/kubernetes/operator/autoscaler/config/AutoScalerOptions.java

```diff
@@ -52,6 +52,13 @@ private static ConfigOptions.OptionBuilder autoScalerConfig(String key) {
                     .defaultValue(Duration.ofMinutes(5))
                     .withDescription("Scaling metrics aggregation window size.");
 
+    public static final ConfigOption<Integer> METRICS_WINDOW_MIN_OBSERVATIONS =
+            autoScalerConfig("metrics.window.min-observations")
+                    .intType()
+                    .defaultValue(10)
```

Comment:
I think the default value here is inconsistent with the default 5 minute metric window. I suggest setting the default here to 5

Reply:
Or we could compute a dynamic default here by computing `(metricWindow / reconcileInterval) - X`

Reply:
True, the metric collection interval is coupled with the reconciliation interval (60 seconds by default). If we chose to calculate the min observations dynamically, I wonder whether we should even expose this as a config option then. It is more of a sanity check.

Reply:
We could also define the config in terms of % of the maximum observations possible (based on the other configs).